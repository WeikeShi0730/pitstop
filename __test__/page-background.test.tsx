import { render, RenderResult } from "@testing-library/react";
import PageBackground from "../components/page-background.component";

let documentBody: RenderResult;
describe("<PageBackground />", () => {
  beforeEach(
    () =>
      (documentBody = render(
        <PageBackground
          info={{ teamBackgrounds: ["test"], fullname: "test name" }}
        />
      ))
  );
  it("Test <PageBackground />", () => {
    expect(documentBody.queryByText("test name")).toBeInTheDocument();
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
