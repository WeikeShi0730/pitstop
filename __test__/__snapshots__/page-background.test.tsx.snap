// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageBackground /> Test <PageBackground /> 1`] = `
<body>
  <div>
    <div
      class="relative bg-slate-100 flex p-10 shadow-2xl hover:shadow-md hover:shadow-slate-500 transition-all ease-in-out duration-200"
    >
      <div
        class="flex relative w-full h-96 2xl:h-100"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="background image"
            class="object-cover rounded-lg"
            data-nimg="fill"
            decoding="async"
            src="test"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </span>
        <div
          class="w-full h-96 2xl:h-100 flex justify-center md:justify-end z-10"
        >
          <div
            class="w-full md:w-1/2 h-full flex justify-center md:justify-end items-center md:bg-gradient-to-l from-slate-50 via-slate-50 opacity-95"
          >
            <div
              class="w-1/2 flex justify-center text-center md:text-right text-xl md:text-3xl text-slate-100 md:text-slate-700 bg-slate-700 p-3 rounded-lg backdrop-blur-md bg-opacity-80 md:backdrop-blur-none md:bg-opacity-100 md:bg-transparent hover:text-slate-100 hover:bg-slate-500 hover:backdrop-blur-md hover:shadow-lg md:hover:text-slate-500 md:hover:bg-transparent md:hover:backdrop-blur-none md:hover:bg-opacity-100 md:hover:shadow-none "
            >
              test name
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
